cartography/graph/cleanupbuilder.py

```diff
@@ -1,61 +1,38 @@
 from dataclasses import asdict
 from string import Template
 from typing import List
-from typing import Optional
-from typing import Set
 
 from cartography.graph.querybuilder import _build_match_clause
-from cartography.graph.querybuilder import filter_selected_relationships
 from cartography.graph.querybuilder import rel_present_on_node_schema
 from cartography.models.core.nodes import CartographyNodeSchema
 from cartography.models.core.relationships import CartographyRelSchema
 from cartography.models.core.relationships import LinkDirection
 from cartography.models.core.relationships import TargetNodeMatcher
 
 
-def build_cleanup_queries(
-        node_schema: CartographyNodeSchema,
-        selected_rels: Optional[Set[CartographyRelSchema]] = None,
-) -> List[str]:
+def build_cleanup_queries(node_schema: CartographyNodeSchema) -> List[str]:
     """
     Generates queries to clean up stale nodes and relationships from the given CartographyNodeSchema.
-    :param node_schema: The given CartographyNodeSchema to generate cleanup queries for.
-    :param selected_rels: Optional. If specified, only generate cleanup queries where the `node_schema` is bound to this
-    given set of selected relationships. Raises an exception if any of the rels in `selected_rels` aren't actually
-    defined on the `node_schema`.
-    If `selected_rels` is not specified (default), we generate cleanup queries against all relationships defined on the
-    `node_schema`.
-    :return: A list of Neo4j queries to clean up nodes and relationships. Order matters: we always clean up the sub
-    resource relationship last because we only clean up stale nodes and rels that are associated with a given sub
-    resource, so if we delete the sub resource first then we will not be able to reach the stale nodes and rels, thus
-    leaving orphaned objects behind.
-    Note also that we return the empty list if the node_schema has no relationships. Doing cleanups of nodes without
-    relationships can be resource expensive for a large graph, and you might risk deleting unintended objects. Please
-    write a manual cleanup job if you wish to do this.
+    Note that auto-cleanups for a node with no relationships is not currently supported.
+    Algorithm:
+    1. First delete all stale nodes attached to the node_schema's sub resource
+    2. Delete all stale node to sub resource relationships
```

Comment:
I'd like a comment here to better explain why this is needed. Something about how the node may exist but is somehow part of another sub resource.